CRM/Core/BAO/CustomField.php

```diff
@@ -2167,6 +2157,24 @@ public static function customOptionGroup($includeFieldIds = NULL) {
     return $customOptionGroup[$cacheKey];
   }
 
+  /**
+   * Get defaults for new entity.
+   *
+   * @return array
+   */
+  public static function getDefaults() {
+    return [
+      'is_required' => FALSE,
+      'is_searchable' => FALSE,
+      'in_selector' => FALSE,
+      'is_search_range' => FALSE,
+      //CRM-15792 - Custom field gets disabled if is_active not set
+      // this would ideally be a mysql default.
+      'is_active' => TRUE,
+      'is_view' => FALSE,
```

Comment:
Why wouldn't all of these be mysql defaults?

Reply:
Yeah beats me - but I wasn't going that far in this PR